cpp-inference/inference_scaling_imagenet.cpp

```diff
@@ -152,32 +158,37 @@ void run_mnist(const std::string& model_name,
     put_tensor_times.push_back(delta_t);
 
     double run_script_start = MPI_Wtime();
-    client.run_script(script_key, "pre_process_3ch", {in_key}, {script_out_key});
+    if (use_multi)
+      client.run_script_multigpu(script_key, "pre_process_3ch", {in_key}, {script_out_key}, rank, num_devices);
+    else
+      client.run_script(script_key, "pre_process_3ch", {in_key}, {script_out_key});
     double run_script_end = MPI_Wtime();
     delta_t = run_script_end - run_script_start;
     run_script_times.push_back(delta_t);
 
     double run_model_start = MPI_Wtime();
-    client.run_model(model_key, {script_out_key}, {out_key});
+    if (use_multi)
+      client.run_model_multigpu(model_key, {script_out_key}, {out_key}, rank, num_devices);
+    else
+      client.run_model(model_key, {script_out_key}, {out_key});
     double run_model_end = MPI_Wtime();
     delta_t = run_model_end - run_model_start;
     run_model_times.push_back(delta_t);
 
-  double unpack_tensor_start = MPI_Wtime();
-  client.unpack_tensor(out_key, result, {1,1000},
-		                   SRTensorTypeFloat,
-                       SRMemLayoutNested);
-  double unpack_tensor_end = MPI_Wtime();
-  delta_t = unpack_tensor_end - unpack_tensor_start;
-  unpack_tensor_times.push_back(delta_t);
+    double unpack_tensor_start = MPI_Wtime();
+    client.unpack_tensor(out_key, result, {1,1000},
+			 SRTensorTypeFloat,
+			 SRMemLayoutNested);
+    double unpack_tensor_end = MPI_Wtime();
+    delta_t = unpack_tensor_end - unpack_tensor_start;
+    unpack_tensor_times.push_back(delta_t);
 
   }
   double loop_end = MPI_Wtime();
   delta_t = loop_end - loop_start;
 
   // write times to file
-  for (int i=0; i<100; i++) {
-
+  for (int i = 1; i < 101; i++) { // Skip first run as it's warmup
```

Comment:
ok I see the reason now. would it be better to invoke this once outside the main loop? I feel like we could re-use this to do it if we made this part into a function with the number of iterations as an argument. either way I understand this now so disregard previous question

Reply:
We probably should add config settings for number of warmup runs and number of test runs, though that's probably outside the scope of this change.

As far as moving it to a separate function, I worry about code duplication.